### PR TITLE
[CustomDevice] fix broadcast when input tensor on CPU 

### PR DESCRIPTION
**Pull request by YanhuiDua**
### PR types
Bug fixes

### PR changes
Others 

### Description
 fix broadcast when input tensor on CPU

### CODE REVIEW

**Review by ronny1996 — DISMISSED** [review]
LGTM

**Comment by KimYannn** [comment]
cc: @yangulei  @jinyouzhi

File: paddle/fluid/distributed/collective/process_group_custom.cc [hunk 1]
```diff
@@ -436,6 +431,28 @@ std::shared_ptr<ProcessGroup::Task> ProcessGroupCustom::Broadcast(
           const phi::stream::Stream& stream) {
         int root = opts.source_rank * in_wrapper.size() + opts.source_root;
         if (rank_ == root) {
+          if (input.place().GetType() == phi::AllocationType::CPU) {
+            platform::CustomPlace place(device_type_, rank_);
```
Comment (yangulei) [inline paddle/fluid/distributed/collective/process_group_custom.cc]: Is it always true that `rank_ == device_id`?

File: paddle/fluid/distributed/collective/process_group_custom.cc [hunk 2]
```diff
@@ -436,6 +431,28 @@ std::shared_ptr<ProcessGroup::Task> ProcessGroupCustom::Broadcast(
...
+            auto allocator = std::unique_ptr<phi::Allocator>(
+                new paddle::experimental::DefaultAllocator(place));
+            phi::DenseTensorMeta meta(input.dtype(), input.dims());
+            phi::DenseTensor input_tmp{allocator.get(), meta};
+
+            int64_t numel = input.numel() / size_;
```
Comment (yangulei) [inline paddle/fluid/distributed/collective/process_group_custom.cc]: Why do we need `/ size_` here?

Comment (YanhuiDua) [reply paddle/fluid/distributed/collective/process_group_custom.cc]: Thx for comments. `/ size_` is not needed in broadcast. fixed.

File: paddle/fluid/distributed/collective/process_group_custom.cc (hunk 1 above)
Comment (YanhuiDua) [reply paddle/fluid/distributed/collective/process_group_custom.cc]: fixed, thx

**Pull request closed without merging**